### ISSUE TITLE
Tessel List is inconsistent with which USB and LAN Tessels are available

### ISSUE THREAD

**Issue by johnnyman727** [tessel/t2-cli#180]
I've been noticing that sometimes Tessels that are plugged in via USB or available with LAN (I can see the yellow connection light) aren't showing up with `t2 list`.

**Comment by johnnyman727** [comment]
One way to reproduce the USB devices missing is to try to call `t2 run PATH_TO_CODE` and then cntl+c out of it before it finishes. Then run `t2 list` and the USB connection won't show up. 

Additionally, the `usb` node library takes about 11 seconds to load after I perform this test:
```.js
  console.log('starting');
  var a = new Date();
  var usb = require('usb');
  console.log('done', (new Date()-a) / 1000);
```
@kevinmehall any idea why this could be?

**Comment by johnnyman727** [comment]
The LAN inconsistency is definitely due to a timing issue. Currently, the `list` command prints out devices that are found over the course of 3 seconds. The devices are usually found within the 3 seconds, but the SSH connection isn't always completely opened by the completion of the three seconds. We try to connect in order to test whether or not that Tessel was provisioned already or not. Perhaps the best way get fix this one is to keep track of these 'pending' connections and wait for them to complete.

**Comment by johnnyman727** [comment]
A simple workaround for now that seems to work consistently is to set the `timeout` flag:
```t2 run {PATH} --timeout 5```

**Comment by johnnyman727** [comment]
Fixed by #269.
